prov/efa/src/efa_mr.c

```diff
@@ -234,6 +234,18 @@ static int efa_mr_dereg_impl(struct efa_mr *efa_mr)
 	int err;
 
 	efa_domain = efa_mr->domain;
+
+	if (efa_mr->peer.iface == FI_HMEM_CUDA) {
+#ifdef HAVE_GDRCOPY
```

Comment:
The reg/dereg also seems like it should be in the HMEM code. You can require that providers call a reg/dereg function for `FI_HMEM_CUDA`, it'll be a no-op when cudaMemcpy is being used.

Reply:
Not sure I understand, ofi_gdrcopy_reg are already in HMEM code. Do you mean introduce a function ofi_hmem_reg and ofi_hmem_dereg?

Reply:
Yeah, something like that so providers have a single entry point and don't have to add this logic/ifdefs to use gdrcopy. Let's see what Sean thinks.

Reply:
@iziemba had the same suggestion when reviewing https://github.com/ofiwg/libfabric/pull/6218, so I added a two new hmem ops: ofi_hmem_dev_register and ofi_hmem_dev_deregister.